devito/symbolics/manipulation.py

```diff
@@ -319,12 +319,88 @@ def reuse_if_untouched(expr, args, evaluate=False):
         return expr.func(*args, evaluate=evaluate)
 
 
-def evalmin(a, b):
+def evalrel(func=min, input=None, assumptions=None):
     """
-    Simplify min(a, b) if possible.
+    Simplify evalmin if possible and return nested MIN/MAX expression. This function
+    takes advantage of python's built-in `min`, Sympy's `Min` and our in-house `MIN`
+    to reduce multiple candidates in relation's when possible.
+
+    Parameters
+    ----------
+    func : min or max, the operation to follow
+        Defaults to min
+    input : a list of the candidate symbols to be simplified
+        The candidates to be evaluated. Defaults to None.
+    assumptions : A list of assumptions formed as relationals between candidates,
+        assumed to be True. Defaults to None.
+
+    Examples
+    --------
+    Assuming no values are known for `a`, `b`, `c`, `d` but we know that `d<=a` and
+    `c>=b` we can safely drop `a` and `c` from the candidate list.
+
+    >>> from devito import Symbol
+    >>> a = Symbol('a')
+    >>> b = Symbol('b')
+    >>> c = Symbol('c')
+    >>> d = Symbol('d')
+    >>> evalrel(max, [a, b, c, d], [[Le(d, a), Ge(c, b)]])
+    MAX(a, c)
     """
+    sfunc = (Min if func is min else Max)  # Choose sympy's Min/Max
+    rfunc = (MIN if func is min else MAX)  # Choose built-in MIN/MAX
+
+    mapper = {}
+
+    if assumptions:
+        processed = []
+        # Add relationals if properties satisfy correctness. If `c >= a + b`` and `a` and
+        # `b` satisfy some conditions. e.g (c > a + b) --> [(c > a) and (c > b)] if `a`,
+        # `b` positive
+        for asm in assumptions:
+            if (asm.__class__ in (Ge, Gt)) and asm.rhs.is_Add and asm.lhs.is_positive:
+                if all(i.is_positive for i in asm.rhs.args):
+                    for i in asm.rhs.args:
+                        processed.append(Ge(asm.lhs, i))
```

Comment:
a more pythonic way would be with `extend` and a list comprehension

Reply:
I was always trying to use list comprehension but my limited mind ends in append. So I was failing because I was not using extend.